src/main/java/com/example/a1project/controllers/MainScreenController.java

```diff
@@ -1,16 +1,145 @@
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
 import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.geometry.Insets;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.GridPane;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
 
-public class MainScreenController extends SceneController {
+public class MainScreenController extends SceneController implements Initializable {
+	
+    @FXML
+    private GridPane currentTaskGridPane;
 
+    @FXML
+    private GridPane completedTaskGridPane;
+    
+    ObservableList<Task> currentTasks = FXCollections.observableArrayList();
+	ObservableList<Task> completedTasks = FXCollections.observableArrayList();
+    
 	public void addTask(ActionEvent event) throws IOException {
 		showNewTaskPopup(event);
 	}
 	
 	public void showAllTasks() {
 		
 	}
+	
+	private void newCurrentTaskDisplay(Task task) {
+		
+		AnchorPane anchor = setupTaskItem(task);
+		
+		currentTaskGridPane.add(anchor, 1, currentTaskGridPane.getRowCount());
+	}
+	
+	private void newCompletedTaskDisplay(Task task) {
+		
+		AnchorPane anchor = setupTaskItem(task);
+		
+		completedTaskGridPane.add(anchor, 1, completedTaskGridPane.getRowCount());
+	}
+	
+	private void removeFromCurrentTask(Task task) {
+		int taskIndex = currentTasks.indexOf(task);
+		if (taskIndex == -1) {
+			return;
+		}
+		currentTaskGridPane.getChildren().remove(taskIndex);
+	}
+	
+	
+	
+	private AnchorPane setupTaskItem(Task task){
+		try {
+			FXMLLoader fxmlloader = new FXMLLoader();
+			fxmlloader.setLocation(getClass().getResource("/com/example/a1project/SingleTaskDisplay.fxml"));
+			AnchorPane anchor = fxmlloader.load();
+			
+			SingleTaskDisplayController taskDisplay = fxmlloader.getController();
+			taskDisplay.setData(task);
+			taskDisplay.setMainScreen(this);
+			
+			return anchor;
+		} catch (IOException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	@Override
+	public void initialize(URL arg0, ResourceBundle arg1) {
+//		currentTasks.addListener(new ListChangeListener<Task>() {
+//		    @Override
+//		    public void onChanged(Change<? extends Task> changedItem) {
+//		    	while (changedItem.next()) {
+//		    		if (changedItem.wasUpdated()) {
+//	                    
+//	                }
+//		    	}
+//		    	
+//		    	// TODO: update FXML (VBox or ListView) to new tasks
+//		    	// updateDisplayLists();
+//		    	
+//		        // while (changedItems.next()) to access all items changed
+//		    }});
+//		
+//		completedTasks.addListener(new ListChangeListener<Task>() {
+//		    @Override
+//		    public void onChanged(Change<? extends Task> changedItems) {
+//		    	
+//		    	// TODO: update FXML (VBox or ListView) to new tasks
+//		    	// updateDisplayLists();
+//		    }});
+		
+	}
```

Comment:
Perhaps remove unused code or explain above what the purpose of the code is and when it may be needed in the future. That way future developers will understand your intentions of commenting it out rather than removing it.

Reply:
Will do, this is highly likely to be used in the future.